### PR TITLE
Date sort crash

### PR DESCRIPTION
**Pull request by kugel3**
Fixes the crashes directly related to sorting discussed [here](https://rdxworks.slack.com/archives/C03QFAWBRNX/p1715599065238219). Should also prevent [this crash](https://rdxworks.slack.com/archives/C03QFAWBRNX/p1715611425616339), but it is still unknown what caused it, or how to reproduce it.

To get access to the tests, check out the [Tests](https://github.com/radixdlt/babylon-wallet-ios/pull/1120/commits/bd357f01aa09121c4c554406bfd8d4fd3693ae20) commit.

### CODE REVIEW

**Review by CyonAlexRDX — APPROVED** [review]
LGTM

File: RadixWallet/Features/TransactionHistory/TransactionHistory+Reducer.swift [hunk 1]
```diff
@@ -450,34 +452,38 @@ extension Range {
...
-			if self[id: sectionID] == nil {
-				let month = calendar.startOfMonth(for: day)
-				self[id: sectionID] = .init(day: day, month: month, transactions: transactions.asIdentified())
-			} else {
-				self[id: sectionID]?.transactions.append(contentsOf: transactions)
-			}
-			self[id: sectionID]?.transactions.sort(by: \.time, >)
+			let month = calendar.startOfMonth(for: day)
+			var section = sections[id: sectionID] ?? .init(day: day, month: month, transactions: [])
+			section.transactions.append(contentsOf: transactions)
```
[the remark this answers is not in the document]
Comment (kugel3) [reply RadixWallet/Features/TransactionHistory/TransactionHistory+Reducer.swift:466]: We *are* appending the transactions here. Or do you mean that you want to convert the rhs argument into an identified array first?

[the remark this answers is not in the document]
Comment (matiasbzurovski) [reply RadixWallet/Features/TransactionHistory/TransactionHistory+Reducer.swift:477]: just based on previous [impl](https://github.com/radixdlt/babylon-wallet-ios/pull/1120/files#diff-7baf04a53985312f88025fd2aac34a4ea1d5269d0f372b0a0e87b970e569bfa3L473), which was doing `.asIdentified()` when creating new section:

Screenshot 2024-05-15 at 11 10 53

[the remark this answers is not in the document]
Comment (kugel3) [reply RadixWallet/Features/TransactionHistory/TransactionHistory+Reducer.swift:477]: Furthermore, `append(contentsOf:)` also skips duplicate elements, so creating an empty `IdentifiedArray` and appending elements is functionally equivalent to using `.asIdentified()`.

[the remark this answers is not in the document]
Comment (matiasbzurovski) [reply RadixWallet/Features/TransactionHistory/TransactionHistory+Reducer.swift:477]: thanks for explanation

**Pull request merged**